nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -359,88 +372,100 @@ def fwd_bwd_step(self, dataloader_iter, batch_idx, forward_only):
-        )
-
-        preds_text = []
-        labels_text = []
-        input_text = []
-        for idx, item in enumerate(result['token_ids']):
-            pred = self.tokenizer.ids_to_text(item[batch['context_lengths'][idx] - 1 :])
-            input = self.tokenizer.ids_to_text(item[: batch['context_lengths'][idx] - 1])
-            label = self.tokenizer.ids_to_text(batch['tokens'][idx][batch['context_lengths'][idx] :].tolist())
-            preds_text.append(pred.strip())
-            labels_text.append(label.strip())
-            input_text.append(input.strip())
-
-        metric = self.val_metric[dataloader_idx] if mode == 'validation' else self.test_metric[dataloader_idx]
-        assert len(preds_text) == len(labels_text) == len(input_text)
-        for _, (pred, label) in enumerate(zip(preds_text, labels_text)):
-            # To compute metrics like pearson or spearman correlation, we need to cast the predicted string and labels to floats.
-            pred, label = self.cast_for_metric(
-                pred=pred.strip(),
-                label=label.strip(),
-                metric_name=self.val_metric_name if mode == 'validation' else self.test_metric_name,
-                class_labels=self.cfg.data.validation_ds.metric.get('class_labels', None)
-                if mode == 'validation'
-                else self.cfg.data.test_ds.metric.get('class_labels', None),
-                labels_are_strings=self.cfg.data.validation_ds.metric.get('labels_are_strings', False)
-                if mode == 'validation'
-                else self.cfg.data.test_ds.metric.get('labels_are_strings', False),
+        # loss can be None as super().validation_step returns None when dataloader_iter is exhausted
+        # which can lead to error, adding check to prevent it
+        if not loss is None:  # Ensure its not None
+            outputs = {
+                'loss': loss,
+                'preds': None,
+                'labels': None,
+                'inputs': None,
+            }
+            if mode == 'validation':
+                if type(self.trainer.val_dataloaders) == list and len(self.trainer.val_dataloaders) > 1:
+                    self.validation_step_outputs_sft[dataloader_idx].append(outputs)
+                else:
+                    self.validation_step_outputs_sft.append(outputs)
+            else:
+                if type(self.trainer.test_dataloaders) == list and len(self.trainer.test_dataloaders) > 1:
+                    self.test_step_outputs[dataloader_idx].append(outputs)
+                else:
+                    self.test_step_outputs.append(outputs)
+            return outputs
+            # TODO (sandeepsub): Figure out the subsequent decode bits.
+            length_params: LengthParam = {
+                "min_length": 0,
+                "max_length": batch['tokens'].size(1) - batch['context_lengths'].max(),
+            }
```

Comment:
Seems there's a problem with the formatting since adding the conditional

Reply:
Discussed offline and fixed.